client/state/data-layer/http/actions.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Internal dependencies
+ */
+import { HTTP_REQUEST } from 'state/action-types';
+
+/***
+ * @typedef {Object} RequestDescription
+ * @property {String} url the url to request
+ * @property {String} method the method we should use in the request: GET, POST etc.
+ * @property {Array<Array<String>>} headers array of [ 'key', 'value' ] pairs for the request headers
+ * @property {Object} body data send as body
+ * @property {Boolean} withCredentials save cookie set on request
+ * @property {Object|Function} onSuccess action to dispatch on success with data meta
+ * @property {Object|Function} onFailure action to dispatch on failure with error meta
+ * @property {Object} options other options
+ */
+
+/***
+ * Creates a raw http action request
+ *
+ * @param {RequestDescription} request HTTP request description
+ * @param {?Object} action default action to call on HTTP events
+ * @returns {Object} raw http action
+ */
+export const rawHttp = ( {
```

Comment:
We already have `http` in here https://github.com/Automattic/wp-calypso/blob/ca7d0b78ec0a396dace4486a23bca674dff15e8d/client/state/data-layer/wpcom-http/actions.js#L28. I think we should come up with naming that makes it more clear how both of them differ.